src/Firebase/Auth.php

```diff
@@ -321,29 +347,45 @@ public function createCustomToken($uid, array $claims = null): Token
      */
     public function verifyIdToken($idToken, bool $checkIfRevoked = null, bool $allowFutureTokens = null): Token
     {
+        try {
+            $idToken = $idToken instanceof Token ? $idToken : (new Parser())->parse($idToken);
+        } catch (\Throwable $e) {
+            throw new InvalidArgumentException('The given value could not be parsed as a token: '.$e->getMessage());
+        }
+
         $checkIfRevoked = $checkIfRevoked ?? false;
         $allowFutureTokens = $allowFutureTokens ?? false;
 
-        try {
-            $verifiedToken = $this->idTokenVerifier->verifyIdToken($idToken);
-        } catch (IssuedInTheFuture $e) {
-            if (!$allowFutureTokens) {
-                throw $e;
-            }
-
-            $verifiedToken = $e->getToken();
-        }
+        if ($this->idTokenVerifier instanceof NewIdTokenVerifier) {
+            try {
+                $this->idTokenVerifier->verify($idToken);
+            } catch (Exception\InvalidToken $e) {
```

Comment:
this catch InvalidToken hidden away exception message from inside verifier

for example, expired idToken will be thrown by verifyExpiry with message 'The token is expired since xxx'
with this too broad catch, will result in new InvalidToken without message from line 371